### PR TITLE
Test Pub() and Priv() by exporting the key and signing/verifying from software

### PR DESCRIPTION
**Pull request by chrisfenner**
This PR enhances the crypto_test.go unit-test by adding a test that creates a TPM key, exports it, and then checks that TPM and software can mutually sign/verify one another.

### CODE REVIEW

**Comment by chrisfenner** [comment]
+CC @AlexandreEXFO in case you have comments on this one.

**Comment by AlexandreEXFO** [comment]
@chrisfenner : There are errors in the error messages of the TestRoundTrip function since there are RSA and ECC keys. Otherwise, it's good.

By curiosity, do you plan to make a new version soon with the changes? I would like to stop depending on a fork for a project that uses this library.

**Comment by chrisfenner** [comment]
> @chrisfenner : There are errors in the error messages of the TestRoundTrip function since there are RSA and ECC keys. Otherwise, it's good.
> 
> By curiosity, do you plan to make a new version soon with the changes? I would like to stop depending on a fork for a project that uses this library.

Thank you! I will kick off another release after this PR goes in.

File: tpm2/crypto_test.go [hunk 1]
```diff
@@ -368,3 +373,365 @@ func TestPub(t *testing.T) {
...
+			}, tc.alg, pub, priv)
+		})
+	}
+}
+
+// Uses the `Pub` and `Priv` helpers to import the public and private key.
+// Note that in practice, one could just use `crypto.PrivateKey.Public()` to
+// derive the public key from the private key. This helper uses the `Pub`
+// routine to validate that function specifically while we test the rest.
+func duplicateAndImport(t *testing.T, tpm transport.TPM, h TPMHandle) (crypto.PublicKey, crypto.PrivateKey) {
```
Comment (jkl73) [inline tpm2/crypto_test.go:542]: Seems like this function duplicates and exports the key, but it never "import" the key back to the TPM, maybe need to updat e the function name?

**Review by jkl73 — APPROVED** [review]

File: tpm2/crypto_test.go (hunk 1 above)
Comment (chrisfenner) [reply tpm2/crypto_test.go:542]: Oh yeah, I'll clarify. It "imports" it to software :)

**Comment by chrisfenner** [comment]
> > @chrisfenner : There are errors in the error messages of the TestRoundTrip function since there are RSA and ECC keys. Otherwise, it's good.
> > By curiosity, do you plan to make a new version soon with the changes? I would like to stop depending on a fork for a project that uses this library.
> 
> Thank you! I will kick off another release after this PR goes in.

https://github.com/google/go-tpm/releases/tag/v0.9.3

**Pull request merged**